bindings/java/src/operator.rs

```diff
@@ -428,6 +431,88 @@ async fn do_rename(op: &mut Operator, source_path: String, target_path: String)
     Ok(op.rename(&source_path, &target_path).await?)
 }
 
+/// # Safety
+///
+/// This function should not be called before the Operator are ready.
+#[no_mangle]
+pub unsafe extern "system" fn Java_org_apache_opendal_Operator_removeAll(
+    mut env: JNIEnv,
+    _: JClass,
+    op: *mut Operator,
+    path: JString,
+) -> jlong {
+    intern_remove_all(&mut env, op, path).unwrap_or_else(|e| {
+        e.throw(&mut env);
+        0
+    })
+}
+
+fn intern_remove_all(env: &mut JNIEnv, op: *mut Operator, path: JString) -> Result<jlong> {
+    let op = unsafe { &mut *op };
+    let id = request_id(env)?;
+
+    let path = jstring_to_string(env, &path)?;
+
+    unsafe { get_global_runtime() }.spawn(async move {
+        let result = do_remove_all(op, path).await;
+        complete_future(id, result.map(|_| JValueOwned::Void))
+    });
+
+    Ok(id)
+}
+
+async fn do_remove_all(op: &mut Operator, path: String) -> Result<()> {
+    Ok(op.remove_all(&path).await?)
+}
+
+/// # Safety
+///
+/// This function should not be called before the Operator are ready.
+#[no_mangle]
+pub unsafe extern "system" fn Java_org_apache_opendal_Operator_list(
+    mut env: JNIEnv,
+    _: JClass,
+    op: *mut Operator,
+    path: JString,
+) -> jlong {
+    intern_list(&mut env, op, path).unwrap_or_else(|e| {
+        e.throw(&mut env);
+        0
+    })
+}
+
+fn intern_list(env: &mut JNIEnv, op: *mut Operator, path: JString) -> Result<jlong> {
+    let op = unsafe { &mut *op };
+    let id = request_id(env)?;
+
+    let path = jstring_to_string(env, &path)?;
+
+    unsafe { get_global_runtime() }.spawn(async move {
+        let result = do_list(op, path).await;
+        complete_future(id, result.map(JValueOwned::Object))
+    });
+
+    Ok(id)
+}
+
+async fn do_list<'local>(op: &mut Operator, path: String) -> Result<JObject<'local>> {
+    let obs = op.list(&path).await?;
+
+    let mut env = unsafe { get_current_env() };
+    let jarray = env.new_object_array(
+        obs.len() as jsize,
+        "org/apache/opendal/Entry",
+        JObject::null(),
+    )?;
+
+    for (idx, entry) in obs.iter().enumerate() {
+        let entry = make_entry(&mut env, entry.to_owned(), Metakey::Mode.into())?;
```

Comment:
This entry appears incorrect. We could consider renaming and exposing `Metadata::bit()` API to `Metadata::metakey()`, enabling users to retrieve previously fetched metadata. A new PR can be initiated for this change.